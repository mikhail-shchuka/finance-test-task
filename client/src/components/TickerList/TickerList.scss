@mixin ticker {
  background-color: #eee;
  display: flex;
  justify-content: space-between;
  width: max-content;
  height: 40px;
  padding: 10px;
  border-radius: 10px;
  border: 1px solid #777;
  gap: 10px;
}

.tickers {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0;
  gap: 40px;

  &__item {
    @include ticker();
    list-style: none;
    position: relative;
    background-color: #ccc;
    border-color: transparent;

    &:hover {
      .tickers__delete-ticker {
        display: block;
      }
    }
  }

  &__new-ticker {
    background-color: #eee;
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: max-content;
    height: 40px;
    padding: 10px;
    border-radius: 10px;
    border: 1px solid #777;
    box-sizing: content-box;
    gap: 10px;

    &:hover {
      background-color: #ccc;
      border-color: transparent;
    }

    &:active {
      transform: scale(1.05);
    }
  }

  &__delete-ticker {
    display: none;
    position: absolute;
    top: -12px;
    right: -12px;
    height: 20px;
    width: 20px;
    text-align: center;
    padding: 0;
    border: none;
    background: none;
    border-radius: 50%;
    background-image: url("../../icons/cross-in-cloud.png");
    background-size: cover;

    &:hover {
      border: 2px solid #777;
    }

    &:active {
      border: none;
      background-color: #777;
    }
  }

  &__name {
    padding-bottom: 5px;
    text-align: center;
    font-weight: bold;
  }

  &__price {
    text-align: center;
    color: #137333;
    font-weight: bold;

    &--down {
      color: #A40C0C;
    }
  }

  &__new-title {
    height: 100%;
  }

}

.icon {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;

  &__img {
    background-color: #B1DDBE;
    border-radius: 50%;

    &--down {
      transform: rotate(180deg);
      background-color: #FFA3A3;
    }
  }
}

.option-enter {
  opacity: 0;
}
.option-enter-active {
  opacity: 1;
  transition: opacity 1000ms;
}
.option-exit {
  opacity: 0;
}
.option-exit-active {
  opacity: 1;
  transition: opacity 1000ms;
}