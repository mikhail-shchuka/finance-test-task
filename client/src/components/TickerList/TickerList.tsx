import { ListItem } from '../ListItem';
import { NewTicker } from '../NewTicker';
import './TickerList.scss';
import { useTickersData } from '../../hooks/useTickersData';
import { getSignPrice } from '../../utils/getSignPrice';

export const TickerList = () => {
  const { hasTickers, prevTickers, tickers } = useTickersData();

  if (!hasTickers) {
    return <NewTicker />;
  }

  return (
    <ul className="tickers">
      {tickers.map(ticker => {
        const sign = getSignPrice(prevTickers, ticker);

        return <ListItem key={ticker.ticker} ticker={ticker} sign={sign} />;
      })}
      <NewTicker />
    </ul>
  );
};
