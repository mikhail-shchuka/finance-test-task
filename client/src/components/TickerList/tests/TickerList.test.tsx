import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { mockTicker, mockPrevTicker } from './constants';
import { TickerList } from '../TickerList';

const mockNewTickerTestId = 'newTicker-test-id';
const mockListItemTestId = 'listItem-test-id';

jest.mock('../../../hooks/useTickersData', () => ({
  useTickersData: () => (
    { tickers: [mockTicker], prevTickers: [mockPrevTicker], hasTickers: true }),
}));

jest.mock('../../ListItem', () => {
  return {
    ListItem: () => (<div data-testid={mockListItemTestId} />),
  };
});

jest.mock('../../NewTicker', () => {
  return {
    NewTicker: () => (<div data-testid={mockNewTickerTestId} />),
  };
});

describe('TickerList', () => {
  it('should match with snapshot', () => {
    const { baseElement } = render(<TickerList />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should be rendered necessary components if tickers not null', () => {
    render(<TickerList />);
    expect(screen.getByTestId(mockNewTickerTestId)).toBeInTheDocument();
    expect(screen.getByTestId(mockListItemTestId)).toBeInTheDocument();
  });
});
