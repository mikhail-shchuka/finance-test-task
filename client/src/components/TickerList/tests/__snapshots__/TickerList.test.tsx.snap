// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickerList should match with snapshot 1`] = `
<body>
  <div>
    <ul
      class="tickers"
    >
      <div
        data-testid="listItem-test-id"
      />
      <div
        data-testid="newTicker-test-id"
      />
    </ul>
  </div>
</body>
`;
